builtins/src/types/primitives.rs

```diff
@@ -0,0 +1,83 @@
+use crate::types::SlFrom;
+use compile_state::state::SloshVm;
+use slvm::{VMResult, Value};
+
+impl SlFrom<&Value> for bool {
+    fn sl_from(value: &Value, _vm: &mut SloshVm) -> VMResult<Self> {
+        match value {
+            Value::True => Ok(true),
+            Value::False => Ok(false),
+            Value::Nil => Ok(false),
+            //TODO XXX should undefined be included here?
+            Value::Undefined => Ok(false),
```

Comment:
Is undefined false?

Reply:
Yes or it should panic.  Undefined is an internal state that the user should never see (may panic in cases if it surfaces- this indicates a compiler failure most likely).  It is used to initialize and clear the stack.  The real driver is that closed over values, if not cleared from the stack, can lead to random behavior when new code starts using random closed values...

The general idea is that properly compiled bytecode will always initialize a variable before accessing it and accessing an Undefined indicates this did not happen.  Of course "access" needs to constrained to something a bytecode did since it is part Value in some cases it is fine to access...

Ideally, without closed over values, we could just leave whatever junk was in a register there and replace as needed but we can't so this what I came up with.  I am open to simplifying this stuff and periodically think it.